3.3/examples/django/run-in-app-container.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+# Use this script to run one-off commands inside a container of a frontend pod
+# (where you application code lives in)
+#
+# Examples:
+# ./run-in-app-container.sh date
+# ./run-in-app-container.sh env
+# ./run-in-app-container.sh ./manage.py migrate
+
+quoted_args="$(printf " %q" "$@")"
+osc exec -p $(osc get pods -l name=frontend -t '{{ with index .items 0 }}{{ .metadata.name }}{{ end }}') -it -- bash -c "cd \$HOME && scl enable python33 \"$quoted_args\""
```

Comment:
On Mon, Apr 27, 2015 at 12:28 PM, Michal Fojtik <notifications@github.com>
wrote:

> In 3.3/examples/django/run-in-app-container.sh
> <https://github.com/openshift/sti-python/pull/8#discussion_r29135774>:
>
> > @@ -0,0 +1,12 @@
> > +#!/bin/bash
> > +
> > +# Use this script to run one-off commands inside a container of a frontend pod
> > +# (where you application code lives in)
> > +#
> > +# Examples:
> > +# ./run-in-app-container.sh date
> > +# ./run-in-app-container.sh env
> > +# ./run-in-app-container.sh ./manage.py migrate
> > +
> > +quoted_args="$(printf " %q" "$@")"
> > +osc exec -p $(osc get pods -l name=frontend -t '{{ with index .items 0 }}{{ .metadata.name }}{{ end }}') -it -- bash -c "cd \$HOME && scl enable python33 \"$quoted_args\""
>
> just a note, you should not do osc exec to perform migration, but the
> migration should happen automatically when you create the app or you do an
> update to your application source code.
>
I partially agree with the migration, but still having such helper command
does makes sense and will be useful instead of doing osc get pods etc... On
the other hand this could be the same as with osc log, pick first pod and
run exec, if you want more fine grained control you still can, just a
thought :wink:.

> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/sti-python/pull/8/files#r29135774>.
>